fluent-syntax/src/visitor.js

```diff
@@ -0,0 +1,58 @@
+import { BaseNode } from "./ast";
+
+/*
+ * Abstract Visitor pattern
+ */
+export class Visitor {
+  visit(node) {
+    if (Array.isArray(node)) {
+      node.forEach(child => this.visit(child));
+      return;
+    }
+    if (!(node instanceof BaseNode)) {
+      return;
+    }
+    const visit = this[`visit_${node.type}`] || this.genericVisit;
```

Comment:
After some more thought, as a consumer of this API I'd find it strange that the underscore is required here. Plus, it might trigger linter warnings, e.g. https://eslint.org/docs/rules/camelcase. I'm not going to block on this but I'd prefer `visit${node.type}`, e.g. `visitTextElement`.